.github/workflows/lein-test.yaml

```diff
@@ -0,0 +1,41 @@
+name: lein_test
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths: ['src/**','test/**']
+  pull_request:
+    types: [opened, reopened, edited, synchronize]
+    paths: ['src/**','test/**']
+
+jobs:
+  run-lein-tests:
+    name: lein test - Java ${{ matrix.java }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '11', '17' ]
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Setup java
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: ${{ matrix.java }}
+      - name: Install Clojure tools
+        uses: DeLaGuardo/setup-clojure@10.2
+        with:
+          cli: latest              # Clojure CLI based on tools.deps
+          lein: latest                  # Leiningen
+          boot: latest                  # Boot.clj
+          bb: latest                    # Babashka
+          clj-kondo: latest        # Clj-kondo
+          cljstyle: latest             # cljstyle
+          zprint: latest                # zprint
+      - name: Run lein tests with dev test profile
+        run: lein test
+      - name: Run lein tests with pseudo-dev, fips profiles
```

Comment:
We can remove this entry (and the name on the one above can be changed to 'Run lein tests')